tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -2980,3 +2981,134 @@ def test_9_barbican(self):
+    def test_10_barbican(self):
+        try:
+            test_var = "tempest.api.workloadmgr.barbican.test_"
+            tests = [[test_var + "create_full_snapshot_with_encrypted_and_unencrypted_volume_instance", 0]]
+            reporting.add_test_script(tests[0][0])
+
+            #create key pair...
+            self.kp = self.create_key_pair(tvaultconf.key_pair_name)
+
+            #create vm...
+            vm_id = self.create_vm(key_pair=self.kp)
+
+            # find volume_type = encrypted and unencrypted.
+            # Get the volume_type_id
+            encrypted_vol_type = -1
+            unencrypted_vol_type = -1
+            for vol in CONF.volume.volume_types:
+                if (vol.lower().find("luks") != -1):
+                    encrypted_vol_type = CONF.volume.volume_types[vol]
+                else:
+                    unencrypted_vol_type = CONF.volume.volume_types[vol]
+
+            if (encrypted_vol_type == -1):
+                reporting.add_test_step("Encrypted volume type is missing." \
+                                        "openstack-setup.conf should have luks volume types like - luks-lvm, luks-ceph, luks etc." \
+                                        "Cannot continue with the test...", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+                raise Exception("No luks volume type found to create encrypted volume.")
+
+            if (unencrypted_vol_type == -1):
+                reporting.add_test_step("Unecrypted volume type is missing." \
+                                        "openstack-setup.conf should have volume types like - lvm, ceph, rbd etc." \
+                                        "Cannot continue with the test...", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+                raise Exception("No volume type like lvm, ceph, rbd etc. found to create unencrypted volume.")
+
+            # Now create unencrypted volume with derived volume type id...
+            volume_id = self.create_volume(
+                volume_type_id=unencrypted_vol_type)
+
+            # Attach volume to vm...
+            self.attach_volume(volume_id, vm_id)
+            LOG.debug("volume id attached : " + str(volume_id))
+
+            # create a workload with encryption status as disabled and try to attach it to vm...
+            wid = self.workload_create([vm_id],
+                                       tvaultconf.workload_type_id, encryption=False,
+                                       workload_cleanup=True)
+
+            LOG.debug("Workload ID: " + str(wid))
+
+            if (wid is not None):
+                self.wait_for_workload_tobe_available(wid)
+                if (self.getWorkloadStatus(wid) == "available"):
+                    reporting.add_test_step("Create unencrypted workload",
+                                            tvaultconf.PASS)
+                    reporting.set_test_script_status(tvaultconf.PASS)
```

Comment:
no need to explicitly set the test script status to PASS. Its value is PASS by default. Only in case of failure, you need to set it to FAIL.